### PR TITLE
Refactoring and Suggestions

### PR DESCRIPTION
**Pull request by gbenattar**
This PR includes changes from https://github.com/KZen-networks/multisig-schnorr/pull/8 with in addition a rebase from master and removal of compiler warnings.

### CODE REVIEW

File: src/protocols/two_party_schnorr/mu_sig/test.rs [hunk 1]
```diff
@@ -1,320 +1,181 @@
...
-                &message,
-                &is_musig
-            ).is_ok()
-        )
+        assert!(EphemeralKey::verify(&ec_context, &s, &r, &party1_key_agg.apk, &message).is_ok())
     }
-
-    #[test]
-
-    fn test_schnorr_one_party() {
```
Comment (omershlo) [inline src/protocols/two_party_schnorr/mu_sig/test.rs]: this test is missing from the pr

File: src/protocols/two_party_schnorr/mu_sig/test.rs [hunk 2]
```diff
@@ -1,320 +1,181 @@
...
-                &party1_key.party1_public_key,
-                &message,
-                &is_musig
-            ).is_ok()
-        )
-    }
-
-    #[test]
-
-    fn test_schnorr_bip_test_vector_2() {
```
Comment (omershlo) [inline src/protocols/two_party_schnorr/mu_sig/test.rs:257]: this test is missing from the pr

File: src/protocols/two_party_schnorr/mu_sig/mod.rs [hunk 3]
```diff
@@ -1,20 +1,264 @@
...
+                c,
+                &BigInt::mod_mul(&x.private_key.to_big_int(), a, &EC::get_q()),
+                &EC::get_q(),
+            ),
+            &EC::get_q(),
+        )
+    }
+}
+
+pub mod party_two {
```
Comment (omershlo) [inline src/protocols/two_party_schnorr/mu_sig/mod.rs]: party 2 is redundent. The only function in this mod is `key_aggregation` which can be replaced by function `key_aggregation_n `. See line 138 in test.rs

**Comment by omershlo** [comment]
@gbenattar see comments above

**Comment by omershlo** [comment]
I completed the refactoring. please review and lets merge this branch

**Comment by gbenattar** [comment]
LGTM, please assign compiler warnings ;) and after I will merge.

**Pull request merged**